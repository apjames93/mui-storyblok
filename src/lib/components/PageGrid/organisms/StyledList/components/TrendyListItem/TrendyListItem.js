/* eslint-disable max-len */
import React, { lazy, Suspense } from 'react';
import PropTypes from 'prop-types';
import Avatar from '@material-ui/core/Avatar';
import { useDynamicAvatarStyles } from '@mui-treasury/styles/avatar/dynamic';
import { useTrendInfoStyles } from '@mui-treasury/styles/info/trend';
import { Column, Row, Item } from '@mui-treasury/components/flex';
import {
  Info,
  InfoTitle,
  InfoSubtitle,
  InfoCaption,
} from '@mui-treasury/components/info';
import { renderComponentsWithBridge } from 'lib/utils/customComponents';

/* istanbul ignore next */
const ButtonRedirect = lazy(() => import('lib/components/PageGrid/molecules/ButtonRedirect/ButtonRedirect'));
/* istanbul ignore next */
const ButtonDownload = lazy(() => import('lib/components/PageGrid/molecules/ButtonDownload/ButtonDownload'));
/* istanbul ignore next */
const IconButtonRedirect = lazy(() => import('lib/components/PageGrid/molecules/IconButtonRedirect/IconButtonRedirect'));
/* istanbul ignore next */
const IconButtonDownload = lazy(() => import('lib/components/PageGrid/molecules/IconButtonDownload/IconButtonDownload'));
/* istanbul ignore next */
const IconButtonDialog = lazy(() => import('lib/components/PageGrid/templates/IconButtonDialog/IconButtonDialog'));
/* istanbul ignore next */
const ButtonDialog = lazy(() => import('lib/components/PageGrid/templates/ButtonDialog/ButtonDialog'));

const components = {
  ButtonRedirect,
  ButtonDownload,
  ButtonDialog,
  IconButtonRedirect,
  IconButtonDownload,
  IconButtonDialog,
};

const TrendyListItem = ({
  src,
  title,
  subtitle,
  iconButton,
  infoText,
  storyblokClass,
  dataBlokC,
  dataBlokUid,
  titleSize,
  subtitleSize,
}) => {
  const avatarStyles = useDynamicAvatarStyles({ size: 48, radius: 6 });
  const icontBtn = iconButton[0];

  return (
    <Column
      gap={2}
      className={storyblokClass}
      data-blok-c={dataBlokC}
      data-blok-uid={dataBlokUid}
    >
      <Row mt={2} alignItems="center">
        <Item position="middle">
          <Avatar
            classes={avatarStyles}
            src={src}
          />
        </Item>
        <Info useStyles={useTrendInfoStyles} style={{ textAlign: 'left' }}>
          <InfoTitle style={{ fontSize: titleSize }}>{title}</InfoTitle>
          <InfoSubtitle style={{ fontSize: subtitleSize }}>{subtitle}</InfoSubtitle>
          <InfoCaption>
            {infoText}
          </InfoCaption>
        </Info>
        <Item position="right">
          {icontBtn && (
          <Suspense fallback={<></>}>
            {renderComponentsWithBridge(components, { ...icontBtn, size: 'small' })}
          </Suspense>
          )}
        </Item>
      </Row>
    </Column>
  );
};

export default TrendyListItem;

TrendyListItem.propTypes = {
  src: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  titleSize: PropTypes.string,
  subtitleSize: PropTypes.string,
  subtitle: PropTypes.string.isRequired,
  infoText: PropTypes.string,
  iconButton: PropTypes.arrayOf(PropTypes.shape()),
  /** storyblok prop for when in editor to allow click bridge */
  dataBlokC: PropTypes.string,
  /** storyblok prop for when in editor to allow click bridge */
  dataBlokUid: PropTypes.string,
  /** storyblok prop for when in editor to allow click bridge */
  storyblokClass: PropTypes.string,
};

TrendyListItem.defaultProps = {
  iconButton: [],
  infoText: '',
  dataBlokC: '',
  dataBlokUid: '',
  storyblokClass: '',
  titleSize: '0.875rem',
  subtitleSize: '0.875rem',
};
