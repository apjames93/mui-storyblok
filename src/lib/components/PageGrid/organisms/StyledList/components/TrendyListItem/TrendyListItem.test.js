import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import TrendyListItem from './TrendyListItem';

function setup() {
  const props = {
    _editable: '<!--#storyblok#{"name": "TrendyListItem", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
    component: 'TrendyListItem',
    title: 'title',
    subtitle: 'subtitle',
    src: 'cool.png',
  };
  const comp = shallow(<TrendyListItem {...props} />);
  return { comp, props };
}

describe('<TrendyListItem />', () => {
  it('renders TrendyListItem', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<TrendyListItem {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
