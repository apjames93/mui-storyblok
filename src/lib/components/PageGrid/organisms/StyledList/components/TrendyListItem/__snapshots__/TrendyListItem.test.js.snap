// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrendyListItem /> snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-5 FlexItem FlexColumn"
  data-blok-c=""
  data-blok-uid=""
>
  <div
    className="MuiBox-root MuiBox-root-6 FlexItem FlexRow"
  >
    <div
      className="MuiBox-root MuiBox-root-7 FlexItem"
    >
      <div
        className="MuiAvatar-root DynamicAvatar-root-1 DynamicAvatar-root-4 MuiAvatar-circle"
      >
        <img
          className="MuiAvatar-img"
          src="cool.png"
        />
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-10 FlexItem Info"
      style={
        Object {
          "textAlign": "left",
        }
      }
    >
      <h6
        className="MuiTypography-root InfoTitle TrendInfo-title-8 MuiTypography-h6"
        style={
          Object {
            "fontSize": "0.875rem",
          }
        }
      >
        title
      </h6>
      <p
        className="MuiTypography-root InfoSubtitle TrendInfo-subtitle-9 MuiTypography-body1"
        style={
          Object {
            "fontSize": "0.875rem",
          }
        }
      >
        subtitle
      </p>
      <p
        className="MuiTypography-root InfoCaption MuiTypography-body1"
      >
        
      </p>
    </div>
    <div
      className="MuiBox-root MuiBox-root-11 FlexItem"
    />
  </div>
</div>
`;
